sklearn/svm/_classes.py

```diff
@@ -189,6 +193,22 @@ class LinearSVC(LinearClassifierMixin, SparseCoefMixin, BaseEstimator):
     [1]
     """
 
+    _parameter_constraints = {
+        "penalty": StrOptions({"l1", "l2"}),
+        "loss": StrOptions({"hinge", "squared_hinge"}),
+        "dual": ["boolean"],
+        "tol": [Real],
```

Comment:
`tol` is a tolerance amount for determining convergence. In this case, it really only is meaningful to have a tolerance value greater than zero, so you'll need to give it actual bounds. For example,

```suggestion
        "tol": [Interval(Real, 0.0, None, closed="left")],
```
where `Interval` is imported from `..utils._param_validation`. Similarly, `C` must be strictly positive and will need a similar construction.

Same applies for `LinearSVR`.